src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs

```diff
@@ -270,13 +270,12 @@ public void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryComplete
                     if (lastChunk != null && lastChunk.Count() > 0)
                     {
                         var discoveredTestsEvent = new DiscoveredTestsEventArgs(lastChunk);
-                        this.OnDiscoveredTests.SafeInvoke(this, discoveredTestsEvent, "DiscoveryRequest.DiscoveryComplete");
                         this.LoggerManager.HandleDiscoveredTests(discoveredTestsEvent);
+                        this.OnDiscoveredTests.SafeInvoke(this, discoveredTestsEvent, "DiscoveryRequest.DiscoveryComplete");
                     }
 
-                    this.OnDiscoveryComplete.SafeInvoke(this, discoveryCompleteEventArgs, "DiscoveryRequest.DiscoveryComplete");
                     this.LoggerManager.HandleDiscoveryComplete(discoveryCompleteEventArgs);
-                    this.LoggerManager.Dispose();
```

Comment:
Where are we disposing now ?

Reply:
In LoggerManager.HandleDiscoveryCompelte